src/operators.jl

```diff
@@ -858,3 +855,23 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+# special case for digraphs
+@traitfn function merge_vertices!(g::::IsDirected, vs::Vector{U})  where U <: Integer
+    vs = sort!(vs, rev=true)
+    v0 = vs[end]
+    @inbounds for v in vs[1:end-1]
+        @inbounds for u in inneighbors(g, v)
+            if isempty(searchsorted(vs, u, rev=true))
```

Comment:
try `searchsortedfirst`:
```
lvs = length(vs)   # outside the loop
...
i = searchsortedfirst(vs, u, rev=true)
if i <= lvs && vs[i] == u:
    add_edge!(g, u, v0)
end
```

Reply:
benchmark results:
```
julia> results["first"]["19980"]
BenchmarkTools.Trial: 
  memory estimate:  14.07 MiB
  allocs estimate:  438724
  --------------
  minimum time:     57.118 ms (0.00% GC)
  median time:      82.128 ms (0.00% GC)
  mean time:        83.859 ms (2.17% GC)
  maximum time:     139.318 ms (8.46% GC)
  --------------
  samples:          56
  evals/sample:     1

julia> results["sorted"]["19980"]
BenchmarkTools.Trial: 
  memory estimate:  8.77 MiB
  allocs estimate:  91844
  --------------
  minimum time:     111.503 ms (0.00% GC)
  median time:      125.888 ms (0.00% GC)
  mean time:        131.465 ms (0.96% GC)
  maximum time:     232.136 ms (0.00% GC)
  --------------
  samples:          35
  evals/sample:     1
```
Bigger tradeoff, probably worth it? I'll commit this change.